core/domain/user_services.py

```diff
@@ -525,14 +587,123 @@ def get_user_settings_by_gae_id(gae_id, strict=False):
+            id exists in the datastore. Defaults to False.
+
+    Returns:
+        UserAuth or None. If the given user_id does not exist and strict
+        is False, returns None. Otherwise, returns the corresponding
+        UserAuth domain object.
+
+    Raises:
+        Exception. The value of strict is True and given user_id does not exist.
+    """
+    if feconf.ENABLE_USER_AUTH_MODEL is True:
+        user_auth = get_users_auth([user_id])[0]
+        if strict and user_auth is None:
+            logging.error('Could not find user with id %s' % user_id)
+            raise Exception('User not found.')
+        return user_auth
+    return None
+
+
+def get_users_auth(user_ids):
+    """Gets domain objects representing the auth information for the given
+    user_ids.
+
+    Args:
+        user_ids: list(str). The list of user_ids to get UserAuth
+            domain objects for.
+
+    Returns:
+        list(UserAuth|None). The UserAuth domain objects corresponding
+        to the given user ids. If the given user_id does not exist, the
+        corresponding entry in the returned list is None.
+    """
+    result = []
+    if feconf.ENABLE_USER_AUTH_MODEL is True:
+        user_auth_models = user_models.UserAuthModel.get_multi(user_ids)
+        for i, model in enumerate(user_auth_models):
+            if user_ids[i] == feconf.SYSTEM_COMMITTER_ID:
+                result.append(UserAuth(
+                    user_id=feconf.SYSTEM_COMMITTER_ID,
+                    gae_id=feconf.SYSTEM_COMMITTER_ID,
+                ))
+            else:
+                result.append(_transform_user_auth(model))
+    return result
+
+
+def get_user_auth_by_gae_id(gae_id, strict=False):
+    """Return the user auth for a single user.
+
+    Args:
+        gae_id: str. The GAE ID of the user.
+        strict: bool. Whether to fail noisily if no user with the given
+            id exists in the datastore. Defaults to False.
+
+    Returns:
+        UserAuth or None. If the given gae_id does not exist and strict
+        is False, returns None. Otherwise, returns the corresponding
+        UserAuth domain object.
+
+    Raises:
+        Exception. The value of strict is True and given gae_id does not exist.
+    """
+    if feconf.ENABLE_USER_AUTH_MODEL is True:
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, gae_id)
+        user_auth = _transform_user_auth(user_auth_model)
+        if strict and user_auth is None:
+            logging.error('Could not find user with id %s' % gae_id)
+            raise Exception('User not found.')
+        return user_auth
+
+    return None
+
+
+def _transform_user_auth(user_auth_model):
+    """Transform user auth storage model to domain object.
+
+    Args:
+        user_auth_model: UserAuthModel. The model to be converted.
+
+    Returns:
+        UserAuth. Domain object for user auth.
+    """
+    if user_auth_model:
```

Comment:
user_auth_model is of type UserAuthModel, so you shouldn't be passing in None, according to the docstring. Therefore this "if" check should not be needed.